network-dashboard-proxy/src/main.rs

```diff
@@ -0,0 +1,253 @@
+            }
+            Error::Network(err) => {
+                tracing::error!("Error processing the stream from the node: {}", err);
+                (
+                    StatusCode::BAD_GATEWAY,
+                    axum::Json("Error processing stream from the node.".into()),
+                )
+            }
+        };
+        r.into_response()
+    }
+}
+
+async fn transaction_status(
+    axum::extract::Path(tx): axum::extract::Path<TransactionHash>,
+    axum::extract::State(mut client): axum::extract::State<v2::Client>,
+) -> Result<axum::Json<TransactionStatus>, Error> {
+    let r = client.get_block_item_status(&tx).await?;
+    Ok(r.into())
+}
+
+async fn blocks_by_height(
+    axum::extract::Path(height): axum::extract::Path<u64>,
+    axum::extract::State(mut client): axum::extract::State<v2::Client>,
+) -> Result<axum::Json<Vec<BlockHash>>, Error> {
+    let _span = tracing::debug_span!("blocks_by_height", height = height);
+    tracing::debug!("Request for blocks at height {height}.");
+    let r = client
+        .get_blocks_at_height(&AbsoluteBlockHeight::from(height).into())
+        .await?;
+    Ok(r.into())
+}
+
+async fn block_info(
+    axum::extract::Path(block_hash): axum::extract::Path<BlockHash>,
+    axum::extract::State(mut client): axum::extract::State<v2::Client>,
+) -> Result<axum::Json<BlockInfo>, Error> {
+    let _span = tracing::debug_span!("block_info");
+    tracing::debug!("Request for block info for {block_hash}.");
+    let r = client.get_block_info(&block_hash).await?;
+    Ok(r.response.into())
+}
+
+async fn consensus_status(
+    axum::extract::State(mut client): axum::extract::State<v2::Client>,
+) -> Result<axum::Json<ConsensusInfo>, Error> {
+    let _span = tracing::debug_span!("consensus_status");
+    tracing::debug!("Request for consensus status.");
+    let r = client.get_consensus_info().await?;
+    Ok(r.into())
+}
+
+async fn block_summary(
+    axum::extract::Path(block_hash): axum::extract::Path<BlockHash>,
+    axum::extract::State(mut client): axum::extract::State<v2::Client>,
+) -> Result<axum::Json<serde_json::Value>, Error> {
+    let _span = tracing::debug_span!("block_summary");
+    tracing::debug!("Request for block summary for {block_hash}.");
+    let txs = client
+        .get_block_transaction_events(block_hash)
+        .await?
+        .response
+        .try_collect::<Vec<_>>()
+        .await?;
+    let special = client
+        .get_block_special_events(block_hash)
+        .await?
+        .response
+        .try_collect::<Vec<_>>()
+        .await?;
+    let finalization_data = client
+        .get_block_finalization_summary(block_hash)
+        .await?
+        .response;
+    let pending_updates = client
+        .get_block_pending_updates(block_hash)
+        .await?
+        .response
+        .try_collect::<Vec<_>>()
+        .await?;
+    let chain_parameters = client
+        .get_block_chain_parameters(block_hash)
+        .await?
+        .response;
```

Comment:
These could be done currently, but not so sure if it is worth the added code complexity.
Just want to mention it.